src/MenuItem/ToggleableItem.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace PhpSchool\CliMenu\MenuItem;
+
+use PhpSchool\CliMenu\MenuItem;
+use PhpSchool\CliMenu\MenuStyle;
+use PhpSchool\CliMenu\Util\StringUtil;
+
+class ToggleableItem implements MenuItem\MenuItemInterface
+{
+    /**
+     * @var callable
+     */
+    private $selectAction;
+
+    /**
+     * @var string
+     */
+    private $text = '';
+
+    /**
+     * @var bool
+     */
+    private $showItemExtra = false;
+
+    /**
+     * @var bool
+     */
+    private $disabled = false;
+
+    /**
+     * @var bool
+     */
+    private $toggled = false;
+
+    public function __construct(
+        string $text,
+        callable $selectAction,
+        bool $showItemExtra = false,
+        bool $disabled = false
+    ) {
+        $this->text          = $text;
+        $this->selectAction  = $selectAction;
+        $this->showItemExtra = $showItemExtra;
+        $this->disabled      = $disabled;
+    }
+
+    /**
+     * Execute the items callable if required
+     */
+    public function getSelectAction() : ?callable
+    {
+        return $this->selectAction;
+    }
+
+    /**
+     * Return the raw string of text
+     */
+    public function getText() : string
+    {
+        return $this->text;
+    }
+
+    /**
+     * Set the raw string of text
+     */
+    public function setText(string $text) : void
+    {
+        $this->text = $text;
+    }
+
+    /**
+     * The output text for the item
+     */
+    public function getRows(MenuStyle $style, bool $toggled = false) : array
```

Comment:
I think `$toggled` is unused here?

Reply:
Yeah, the default `$selected` from other classes would also be unused here.

We can use it for a `is selected but not checked`?

Reply:
Ah yeah I see. I don't think we a selected marker for this type. I guess it needs to be there for the interface but we can just ignore. Unless you want to implement something like selected but not checked, but I'm not sure how that would work.